src/Communciator/CommunicatorEvents.js

```diff
@@ -80,6 +80,17 @@ module.exports = class CommunicatorEvents extends EventEmitter {
       if (data.from.startsWith(this.communicator.client.authenticator.accountId)) return;
       if (data.type === 'unavailable' && !data.delay) return; // should mean it was a failed presence (removed friend)
 
+      let platform = ''
```

Comment:
This can be a constant with the if cases as a x ? a : b instead, since of `this.platform = data.platform || ''`

Reply:
I'm not sure I'm following. do you mean something like this:

      const fromSplit = data.from.split('Fortnite:');

      if (fromSplit && fromSplit.length > 1) {
        const platformSrc = fromSplit[1];

        if (platformSrc.length > 2) {
          data.platform = platformSrc.substr(0, 3);
        }
      }

and then:

    const friend = new Friend(this.communicator, {
      ...
      platform: data.platform || ''
    })

Reply:
or maybe this:

    const fromSplit = data.from.split('Fortnite:');
    const platform = (fromSplit && fromSplit.length) > 1 ? (fromSplit[1].length > 2 ? platformSrc.substr(0, 3) : '') : '';